### PR TITLE
fix: correct parent linking

### PR DESCRIPTION
**Pull request by stijnmoreels**

### CODE REVIEW

File: src/Arcus.POC.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/OperationContextConverter.cs [hunk 1]
```diff
@@ -21,41 +22,71 @@ public class OperationContextConverter
...
+                {
+                    telemetryEntry.Context.Operation.Id = transactionId;
+                }
+
+                if (telemetryEntry.Properties.TryGetValue(ContextProperties.Correlation.OperationId, out string operationId))
+                {
+                    telemetryEntry.Context.Operation.ParentId = operationId;
+                }
             }
+        }
```
Comment (stijnmoreels) [inline src/Arcus.POC.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/OperationContextConverter.cs:55]: This is actually the most major change.

File: src/Arcus.POC.WebApi.Logging.Core/Correlation/HttpCorrelation.cs [hunk 2]
```diff
@@ -109,20 +173,26 @@ public bool TryHttpCorrelate(out string errorMessage)
...
+            //    }
+            //}
+            //else
+            //{
+            //    operationId = DetermineOperationIdWithoutUpstreamService(httpContext);
+            //}
+            // new:
+            operationParentId = httpContext.Request.Headers[_options.UpstreamService.OperationParentIdHeaderName];
+            operationId = DetermineOperationIdWithoutUpstreamService(httpContext);
+            _logger.LogTrace("Correlation: {OperationId}, {TransactionId}, {OperationParentId}", operationId, transactionId, operationParentId);
```
Comment (stijnmoreels) [inline src/Arcus.POC.WebApi.Logging.Core/Correlation/HttpCorrelation.cs:195]: Currently not using this `|abc.def` format and just place the parent ID in the header as-is.

**Comment by stijnmoreels** [comment]
Currently, I'm thinking of these TODO tasks (also documented in the code):
- [ ] Web API middleware should extract only last part of operation parent ID ( `Request-Id`: `|abc.def` should become `Parent ID`: `def` in Application Insights, currently, it's `|abc.def`) (https://github.com/arcus-azure/arcus-service-to-service-correlation-poc/pull/13/files#diff-fb92c0f5cee4ab12e7f9150943d525590093b76916d4e94ba3949ed80a06dd3fR194)
- [ ] Application Insights Sink should differentiate between request and other types of telemetry when assigning correlation, the request telemetry needs the operation ID as it's telemetry ID, while the other types need the operation ID as parent ID.  (https://github.com/arcus-azure/arcus-service-to-service-correlation-poc/pull/13/files#diff-8ca727275995a3807a592ad7bd393c6044e1dc7db434516cda2ffd1d9bf40dd8R28)
- [ ] Message router should assign the correlation directly instead of via the `IMessageCorrelationInfoAccessor` because the Serilog enrichment happens at a later stage, outside the availability of the accessor. (https://github.com/arcus-azure/arcus-service-to-service-correlation-poc/pull/13/files#diff-baa446f63c0f3d982de91acaf8252216a3256b92bdc9a53d66a7e8c42f7901dbR214)

File: src/Arcus.POC.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/OperationContextConverter.cs (hunk 1 above)
Comment (fgheysels) [reply src/Arcus.POC.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/OperationContextConverter.cs:57]: I have seen the screenshot with the results, so I assume this is ok :)

I also assume that this code will need to be added in the Arcus lib itself ?  I would remove the 'generated-' prefix in the final code then.

File: src/Arcus.POC.WebApi.Logging.Core/Correlation/HttpCorrelation.cs (hunk 2 above)
Comment (fgheysels) [reply src/Arcus.POC.WebApi.Logging.Core/Correlation/HttpCorrelation.cs:195]: I currently have no opinion if we should move away from that format or not.

**Review by fgheysels — APPROVED** [review]

File: src/Arcus.POC.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/OperationContextConverter.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.POC.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/OperationContextConverter.cs:57]: 😊 yeah, I added this `generated-abc123` stuff bc I would know when it was used 😅 , I guess that we can reuse the functionality available already in Arcus for generation such ID's, so we'll be fine. 👌

File: src/Arcus.POC.WebApi.Logging.Core/Correlation/HttpCorrelation.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.POC.WebApi.Logging.Core/Correlation/HttpCorrelation.cs:195]: We could change it a bit that we still support the HTTP correlation standard in the communication but not in Application Insights.
I'm even wondering if that was ever even correct as this is HTTP-specific.

**Pull request merged**